### PR TITLE
API: add api_meta to order

### PR DESCRIPTION
**Pull request by wiffbi**
This PR adds api_meta_info to the order-object. This meta-info is only available through the API. A new signal for plugins to write to this meta-object on order-creation is added as well.

### CODE REVIEW

File: doc/api/resources/orders.rst [hunk 1]
```diff
@@ -742,6 +744,8 @@ Updating order fields
 
    * ``comment``
 
+   * ``api_meta_info`` (you always need to supply the full object, or ``null`` to delete the current value)
```
Comment (raphaelm) [inline doc/api/resources/orders.rst:747]:
```suggestion
   * ``api_meta`` (you always need to supply the full object, or ``{}`` to delete the current value)
```

File: src/pretix/base/models/orders.py [hunk 2]
```diff
@@ -299,6 +299,11 @@ class Order(LockModel, LoggedModel):
         verbose_name=_("Meta information"),
         null=True, blank=True
     )
+    api_meta_info = models.JSONField(
+        verbose_name=_("API-Meta information"),
```
Comment (raphaelm) [inline src/pretix/base/models/orders.py:303]:
```suggestion
        verbose_name=_("API meta information"),
```

File: src/pretix/base/models/orders.py [hunk 3]
```diff
@@ -299,6 +299,11 @@ class Order(LockModel, LoggedModel):
...
+        null=True, blank=True,
```
Comment (raphaelm) [inline src/pretix/base/models/orders.py:304]: Why should this be nullable? Do we need the semantic difference between NULL and {}?

```suggestion
        null=False, blank=True,
```

File: doc/api/resources/orders.rst [hunk 4]
```diff
@@ -42,6 +42,7 @@ payment_date                          date                       **DEPRECATED AN
 payment_provider                      string                     **DEPRECATED AND INACCURATE** Payment provider used for this order
 total                                 money (string)             Total value of this order
 comment                               string                     Internal comment on this order
+api_meta_info                         object                     Meta-data for that order. Only available through API.
```
Comment (raphaelm) [inline doc/api/resources/orders.rst:45]: I'm not happy about the naming, see slack discussion

File: src/pretix/base/migrations/0269_order_api_meta_info.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+    dependencies = [
+        ('pretixbase', '0268_remove_subevent_items_remove_subevent_variations_and_more'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='order',
+            name='api_meta_info',
+            field=models.JSONField(default=dict, null=True),
```
Comment (raphaelm) [inline src/pretix/base/migrations/0269_order_api_meta_info.py:16]:
```suggestion
            field=models.JSONField(default=dict, null=False),
```

File: doc/api/resources/orders.rst [hunk 6]
```diff
@@ -742,6 +744,8 @@ Updating order fields
 
    * ``comment``
 
+   * ``api_meta`` (you always need to supply the full object, or ``null`` to delete the current value)
```
Comment (raphaelm) [inline doc/api/resources/orders.rst:747]:
```suggestion
   * ``api_meta``
```

File: doc/api/resources/orders.rst [hunk 7]
```diff
@@ -42,6 +42,7 @@ payment_date                          date                       **DEPRECATED AN
 payment_provider                      string                     **DEPRECATED AND INACCURATE** Payment provider used for this order
 total                                 money (string)             Total value of this order
 comment                               string                     Internal comment on this order
+api_meta                              object                     Meta-data for that order. Only available through API.
```
Comment (raphaelm) [inline doc/api/resources/orders.rst:45]:
```suggestion
api_meta                              object                     Meta data for that order. Only available through API, no guarantees
                                                                 on the content structure. You can use this to save references to your system.
```

**Review by raphaelm — APPROVED** [review]

File: src/pretix/base/migrations/0269_order_api_meta_info.py (hunk 5 above)
Comment (wiffbi) [reply src/pretix/base/migrations/0269_order_api_meta_info.py]: Fixed

**Pull request merged**